rslib/src/collection/backup.rs

```diff
@@ -0,0 +1,100 @@
+// Copyright: Ankitects Pty Ltd and contributors
+// License: GNU AGPL, version 3 or later; http://www.gnu.org/licenses/agpl.html
+
+use std::{
+    fs::{read_dir, remove_file, DirEntry, File},
+    io::Write,
+    path::{Path, PathBuf},
+};
+
+use chrono::prelude::*;
+use itertools::Itertools;
+use zip::{write::FileOptions, CompressionMethod, ZipWriter};
+use zstd;
+
+use crate::prelude::*;
+
+const BACKUP_FORMAT_STRING: &str = "backup-%Y-%m-%d-%H.%M.%S.colpkg";
+/// Backups in the last [NO_THINNING_SECS] seconds are not thinned.
+const NO_THINNING_SECS: u64 = 2 * 24 * 60 * 60;
+/// At most one backup every [THINNING_INTERVAL_SECS] seconds is kept.
+const THINNING_INTERVAL_SECS: i64 = 24 * 60 * 60;
+
+pub fn backup(col_path: &str, out_dir: &str) -> Result<()> {
+    let out_file = File::create(out_path(out_dir))?;
+    let mut zip = ZipWriter::new(out_file);
+    let options = FileOptions::default().compression_method(CompressionMethod::Stored);
+
+    let col_file = File::open(col_path)?;
+    let encoded_col = zstd::encode_all(col_file, 0)?;
+
+    zip.start_file("collection.anki2", options)?;
```

Comment:
Older clients will try to read this file, and they copy it over the existing collection without checking it. I tried with 2.1.15, and after importing a .colpkg with random data in it, it throws a 'collection corrupt' message and the interface hangs. Once restarted, the collection can't be opened and the profile needs to be deleted. :sweat:  Newer Anki versions at least don't hang the interface, but still copy the file into place without checking it. That should get addressed by the import/export work, but we can't fix older clients, so I think it would be better if we avoid the issue by using a new filename inside the zip.

Anki 2.1 already uses collection.anki21, and using that will cause the same problems. How about collection.anki22? We should probably also throw in a 'meta' file with '{"ver": 3}'. If our restore backup/import code checks to make sure the version isn't greater than what it supports, it will allow us to avoid having to use any more filenames in the future.

Reply:
> I think it would be better if we avoid the issue by using a new filename inside the zip.

Yes, I was planning to do that, but wanted to set up the frontend first for easier testing, therefore the PR. Sorry, I should have said so!

I'd think `.anki22` was used by Anki 2.2.x. It doesn't really matter to me, but maybe `.anki215`? Or `.anki21a`?
Adding meta sounds good!

Reply:
21b? Any of them work for me though.